react/utils/pages.ts

```diff
@@ -125,14 +125,29 @@ export function navigate(history: History | null, pages: Pages, options: Navigat
   return false
 }
 
-export function scrollTo(options: ScrollToOptions) {
+export function scrollTo(options: RelativeScrollToOptions) {
+  const { baseElementId } = options
+  const scrollAnchor = baseElementId && document.querySelector(`#${baseElementId}`)
+
+  if (!scrollAnchor) {
+    return polyfillScrollTo(options)
+  }
+
+  const { top, left } = scrollAnchor.getBoundingClientRect()
+  polyfillScrollTo({
+    left: left + window.scrollX + (options.left || 0),
+    top: top + window.scrollY + (options.top || 0)
+  })
+}
+
+function polyfillScrollTo(options: ScrollToOptions) {
```

Comment:
The `polyfill` part is just the thing inside the `catch`, but no one cares, I'm just being pedantic.